velox/exec/tests/PrestoQueryRunnerTest.cpp

```diff
@@ -255,4 +255,122 @@ TEST_F(PrestoQueryRunnerTest, toSql) {
+                    .values({t})
+                    .hashJoin(
+                        {"t0"},
+                        {"u0"},
+                        PlanBuilder(planNodeIdGenerator).values({u}).planNode(),
+                        /*filter=*/"",
+                        {"t0", "t1"},
+                        core::JoinType::kInner)
+                    .planNode();
+    EXPECT_EQ(
+        *queryRunner->toSql(plan),
+        "SELECT t0, t1 FROM t_0 INNER JOIN t_1 ON t0 = u0");
+  }
+
+  // Two joins with a filter.
+  {
+    auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+    auto plan = PlanBuilder(planNodeIdGenerator)
+                    .values({t})
+                    .hashJoin(
+                        {"t0"},
+                        {"u0"},
+                        PlanBuilder(planNodeIdGenerator).values({u}).planNode(),
+                        /*filter=*/"",
+                        {"t0"},
+                        core::JoinType::kLeftSemiFilter)
+                    .hashJoin(
+                        {"t0"},
+                        {"v0"},
+                        PlanBuilder(planNodeIdGenerator).values({v}).planNode(),
+                        "v1 > 0",
+                        {"t0", "v1"},
+                        core::JoinType::kInner)
+                    .planNode();
+    EXPECT_EQ(
+        *queryRunner->toSql(plan),
+        "SELECT t0, v1"
+        " FROM (SELECT t0 FROM t_0 WHERE t0 IN (SELECT u0 FROM t_1))"
+        " INNER JOIN t_3 ON t0 = v0 AND (cast(v1 as BIGINT) > BIGINT '0')");
+  }
+
+  // Three joins.
+  {
+    auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+    auto plan = PlanBuilder(planNodeIdGenerator)
+                    .values({t})
+                    .hashJoin(
+                        {"t0"},
+                        {"u0"},
+                        PlanBuilder(planNodeIdGenerator).values({u}).planNode(),
+                        /*filter=*/"",
+                        {"t0", "t1"},
+                        core::JoinType::kLeft)
+                    .hashJoin(
+                        {"t0"},
+                        {"v0"},
+                        PlanBuilder(planNodeIdGenerator).values({v}).planNode(),
+                        /*filter=*/"",
+                        {"t0", "v1"},
+                        core::JoinType::kInner)
+                    .hashJoin(
+                        {"t0", "v1"},
+                        {"w0", "w1"},
+                        PlanBuilder(planNodeIdGenerator).values({w}).planNode(),
+                        /*filter=*/"",
+                        {"t0", "w1"},
+                        core::JoinType::kFull)
+                    .planNode();
+    EXPECT_EQ(
+        *queryRunner->toSql(plan),
+        "SELECT t0, w1"
+        " FROM (SELECT t0, v1 FROM (SELECT t0, t1 FROM t_0 LEFT JOIN t_1 ON t0 = u0)"
+        " INNER JOIN t_3 ON t0 = v0)"
+        " FULL OUTER JOIN t_5 ON t0 = w0 AND v1 = w1");
+  }
+}
```

Comment:
Can you add a test with a scalar sub query too ?

Reply:
The test case with two joins contain a scalar sub query. The scalar subquery being (SELECT u0 FROM t_1).